File: java/ch01-lists-queues-stack/E01Array.java
/*Programmer: Leslye Hernandez Jimenez
 *        /\_/\
 *       ( o.o )    
 *        > ^ <
 * Array Example
 * * Fecha: 22/01/2025
 * c++ e01_array.cpp -o e01_array.exe
 */

public class E01Array {
    public static void main(String[] args) {
        int[] arr = new int[10];

        arr[0] = 1;
        System.out.println("First element: " + arr[0]);
        
        int[] arr2 = {1, 2, 3, 4};
        System.out.println("Elements in second array: ");
        for (int i = 0; i < arr2.length; i++)
            System.out.print(arr2[i] + " ");
        System.out.println();
        
        try {
            int a = arr2[4];
            int b = arr2[5];
        } catch (Exception e) {
            System.out.println("EXCEPTION: " + e.getMessage());
        }
    }
}

File: cpp/ch1-list-queues-stacks/e01_array.cpp
/*Programmer: Leslye Hernandez Jimenez
 *        /\_/\
 *       ( o.o )    
 *        > ^ <
 * Array Example
 * * Fecha: 22/01/2025
 * c++ e01_array.cpp -o e01_array.exe
 */

// librerias
#include <iostream>
#include <array>
using namespace std;

int main(){
    std::array<int, 10> arr;

    arr[0] = 1;
    std::cout << "First element:" << arr[0] << std::endl;

    std::array<int, 4> arr2 = {1, 2, 3, 4};
    std::cout << "Elements in second array: ";
    for(int i = 0; i < arr.size(); i++)
        std::cout << arr2[i] << " ";

    // exception
    try{
        std::cout << arr2.at(4);
        std::cout << arr2.at(5);
    }
    catch (const out_of_range& ex){
        std::cerr << "EXCEPTION: " << ex.what();
    }
}

